src/ucp/core/ucp_request.inl

```diff
@@ -396,6 +398,19 @@ ucp_request_send_state_advance(ucp_request_t *req,
             /* cppcheck-suppress nullPointer */
             req->send.state.dt        = *new_dt_state;
         }
+
+        if (UCS_STATUS_IS_ERR(status) && (status != UCS_ERR_NO_RESOURCE)) {
```

Comment:
`UCS_ERR_NO_RESOURCE` is not possible here due to the check at the beginning of the function